### PR TITLE
feat(bulletin-board): [bulletin-board] Add click events #2143

### PR DESCRIPTION
**Pull request by wuyiping0628**
# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-vue/blob/dev/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #2143 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Release Notes

- **New Features**
	- Introduced a new interactive bulletin board component with enhanced data structure and user click events.
	- Added support for click events that trigger modals displaying item details.
	- Expanded API with new props for better customization of the bulletin board component.
	- Added a new demo for click event functionality.

- **Bug Fixes**
	- Improved type safety for data properties in the bulletin board component.

- **Documentation**
	- Localized descriptions for the new demo added.

- **Style**
	- Enhanced UI feedback with cursor changes for interactive elements.

- **Tests**
	- Added new test cases to ensure click event functionality works as expected.

### CODE REVIEW

File: examples/sites/demos/apis/bulletin-board.js [hunk 1]
```diff
@@ -87,9 +88,35 @@ export default {
           pcDemo: 'title'
         }
       ],
-      events: [],
+      events: [
+        {
+          name: 'handleClick',
```
Comment (zzcr) [inline examples/sites/demos/apis/bulletin-board.js]: 需要添加版本号信息，说明是什么版本添加的

Comment (zzcr) [inline examples/sites/demos/apis/bulletin-board.js]: handleClick这个名字不太好，表达不出来这个点击事件是干啥的

Comment (wuyiping0628) [reply examples/sites/demos/apis/bulletin-board.js]: 已修改

Comment (wuyiping0628) [reply examples/sites/demos/apis/bulletin-board.js]: 已修改

**Review by zzcr — APPROVED** [review]

**Pull request merged**